### DISCUSSION TITLE
Which mapping function (Haldane, Kosambi, ...) is used in AlphaSimR?

### DISCUSSION THREAD

**Discussion by gregorgorjanc** [gaynorr/AlphaSimR#89] [General]
@gaynorr have you described anywhere which mapping function (Haldane, Kosambi, ...) is used in AlphaSimR?

**Comment by gregorgorjanc** [comment]
Looking at https://github.com/gaynorr/AlphaSimR/blob/18c7574ef7c69953d3c054e8046be8e175d9c13c/src/meiosis.cpp#L56 I see that gamma process is used for sampling chiasmata locations and at https://github.com/gaynorr/AlphaSimR/blob/18c7574ef7c69953d3c054e8046be8e175d9c13c/src/meiosis.cpp#L808 that count-location model is used for simulating gametes.

Correct?

I think it would be nice if we document this, say in `makeCross()`. I am happy to do it, if you give me pointers to key references.

**Comment by gaynorr** [comment]
The quick answer is that AlphaSimR effectively models a genetic map constructed using the Kosambi mapping function.

The longer answer is that the current version of AlphaSimR uses a gamma sprinkling model for recombination. The gamma sprinkling model simulates two pathways for recombination, one with crossover interference and one without. The default parameters currently model all crossovers as coming from the interfering pathway with an interference parameter that approximates the interference assumed by the Kosambi mapping function. The user can freely change these parameters (see the `p` and `v` fields in `SimParam`). However, I don't currently offer any functionality for converting the genetic map so it would be up to the user to manually perform this conversion.

Older versions used a gamma model (what is stated in the publication) and even older versions used a count-location model. The count-location model doesn't model interference, so it is equivalent to assuming the Haldane mapping function. When I first switched to the gamma model, I set the default parameters to continue modeling the Haldane mapping function. I changed these defaults to model the Kosambi mapping function in version 1.0.0.

For anyone that is interested, the introduction of [this paper](https://academic.oup.com/plcell/article/21/12/3915/6096114) by Falque _et al._ provides a good introduction to some of the statistical models for recombination. Karl Broman also has a nice [slide deck](https://www.biostat.wisc.edu/~kbroman/presentations/meiosis_nz.pdf) that covers the gamma and count-location models.

**Comment by gregorgorjanc** [comment]
@gaynorr should we document this somewhere in the package man pages?
